src/test/java/org/fundacionjala/coding/omar/CesarEncryptionTest.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    /**
+     * Test to encode message with negative key.
+     */
+    @Test
+    public void testEncodeWithNegativeKey() {
+        final int key = -3;
+        assertEquals("HOLA", enconde.encodeMessageWithCesarEncryption("KROD", key));
+    }
+
+    /**
+     * Test to encode message with empty message.
+     */
+    @Test
+    public void testEncodeWithEmptyMessage() {
+        final int key = 5;
+        assertEquals("", enconde.encodeMessageWithCesarEncryption("", key));
+    }
+
+    /**
+     * Test to encode message with null message.
+     */
+    @Test
+    public void testEncodeWithNullMessage() {
+        final int key = 5;
+        assertEquals("", enconde.encodeMessageWithCesarEncryption(null, key));
+    }
+
+    /**
+     * Test to encode message with special characters and blank spaces.
+     */
+    @Test
+    public void testEncodeWithSpecialCharactersAndBlankSpace() {
+        final int key = 3;
+        assertEquals("K_ROD", enconde.encodeMessageWithCesarEncryption("H_OLA", key));
+        assertEquals("KROD CRUUR", enconde.encodeMessageWithCesarEncryption("HOLA ZORRO", key));
+    }
+
+    /**
+     * Test to decode message with positive key.
+     */
+    @Test
+    public void testDecodeWithPositiveKey() {
+        final int key = 3;
+        assertEquals("X", enconde.decodeMessageWithCesarEncryption("A", key));
+        assertEquals("W", enconde.decodeMessageWithCesarEncryption("Z", key));
+    }
+
+    /**
+     * Test to decode message with negative key.
+     */
+    @Test
+    public void testDecodeWithNegativeKey() {
+        final int key = -3;
+        assertEquals("FKVG", enconde.decodeMessageWithCesarEncryption("CHSD", key));
+        assertEquals("CCGIV VGI", enconde.decodeMessageWithCesarEncryption("ZZDFS SDF", key));
+    }
+
+    /**
+     * Test to decode message with empty message.
+     */
+    @Test
+    public void testDecodeWithEmptyMessage() {
+        final int key = 3;
+        assertEquals("", enconde.decodeMessageWithCesarEncryption("", key));
+    }
+
+    /**
+     * Test to decode message with null message.
+     */
+    @Test
+    public void testDecodeWithNullMessage() {
+        final int key = 3;
+        assertEquals("", enconde.decodeMessageWithCesarEncryption(null, key));
+    }
+
+    /**
+     * Test to encode message with special character or blanks spaces.
+     */
+    @Test
+    public void testDecodeWithSpecialCharactersAndBlankSpace() {
+        final int key = 3;
+        assertEquals("H_OLA", enconde.decodeMessageWithCesarEncryption("K_ROD", key));
+        assertEquals("HOLA ZORRO", enconde.decodeMessageWithCesarEncryption("KROD CRUUR", key));
+    }
+
+    /**
+     * Test to encode with Vigenere Encryption and String key.
+     */
+    @Test
+    public void testEncodeVigenereEncryption() {
+        assertEquals("VJYPGAOG", enconde.encodeMessageWithVigenereEncryption("JALASOFT", "LIMON"));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "LIMON" 8 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)